src/buf/uninit_slice.rs

```diff
@@ -22,10 +22,42 @@ use core::ops::{
 pub struct UninitSlice([MaybeUninit<u8>]);
 
 impl UninitSlice {
-    pub(crate) fn from_slice(slice: &mut [MaybeUninit<u8>]) -> &mut UninitSlice {
+    /// Creates a `&mut UninitSlice` wrapping slice of uninitialised memory.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bytes::buf::UninitSlice;
+    /// use core::mem::MaybeUninit;
+    ///
+    /// let mut buffer = [MaybeUninit::uninit(); 64];
+    /// let slice = UninitSlice::from_slice_mut(&mut buffer[..]);
+    ///
+    /// let mut vec = Vec::with_capacity(1024);
+    /// let spare: &mut UninitSlice = vec.spare_capacity_mut().into();
+    /// ```
+    pub fn from_slice_mut(slice: &mut [MaybeUninit<u8>]) -> &mut UninitSlice {
         unsafe { &mut *(slice as *mut [MaybeUninit<u8>] as *mut UninitSlice) }
     }
```

Comment:
How about calling the two new public methods `from_uninit_slice` and `from_init_slice`? Or `from_uninit_slice`+`from_slice`?

Reply:
I don't think it _needs_ to be consistent with `from_raw_parts_mut`.

Reply:
Done.